lib/improver/tests/nowcasting/lightning/test_NowcastLightning.py

```diff
@@ -0,0 +1,946 @@
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Unit tests for the nowcast.lightning.NowcastLightning plugin."""
+
+
+import unittest
+
+from iris.util import squeeze
+from iris.coords import DimCoord, CellMethod
+from iris.cube import Cube, CubeList
+from iris.tests import IrisTest
+from iris.exceptions import CoordinateNotFoundError, ConstraintMismatchError
+import numpy as np
+import cf_units
+
+from improver.nowcasting.lightning import NowcastLightning as Plugin
+from improver.tests.nbhood.nbhood.test_BaseNeighbourhoodProcessing import (
+    set_up_cube, set_up_cube_with_no_realizations)
+from improver.tests.ensemble_calibration.ensemble_calibration.helper_functions\
+    import add_forecast_reference_time_and_forecast_period
+
+
+class Test__init__(IrisTest):
+
+    """Test the __init__ method accepts keyword arguments."""
+
+    def test_with_radius(self):
+        """
+        Test that the radius keyword is accepted.
+        """
+        radius = 20000.
+        plugin = Plugin(radius=radius)
+        self.assertEqual(plugin.radius, radius)
+
+
+class Test__repr__(IrisTest):
+
+    """Test the repr method."""
+
+    def test_basic(self):
+        """Test that the __repr__ returns the expected string."""
+        # Have to pass in a lambda to ensure two strings match the same
+        # function address.
+        def local_function(mins):
+            """To ensure plugin lambda function is expressed fairly in repr."""
+            return lambda mins: mins
+        set_lightning_thresholds = local_function
+        plugin = Plugin()
+        plugin.lrt_lev1 = set_lightning_thresholds
+        result = str(plugin)
+        msg = ("""<NowcastLightning: radius={radius},
+ lightning mapping (lightning rate in "min^-1"):
+   upper: lightning rate {lthru} => min lightning prob {lprobu}
+   lower: lightning rate {lthrl} => min lightning prob {lprobl}
+>""".format(radius=10000.,
+            lthru=set_lightning_thresholds, lthrl=0.,
+            lprobu=1., lprobl=0.25)
+              )
+        self.assertEqual(result, msg)
+
+
+class Test__update_metadata(IrisTest):
+
+    """Test the _update_metadata method."""
+
+    def setUp(self):
+        """Create a cube with a single non-zero point like this:
+     precipitation_amount / (kg m^-2)
+     Dimension coordinates:
+        realization: 1;
+        time: 1;
+        projection_y_coordinate: 16;
+        projection_x_coordinate: 16;
+     Auxiliary coordinates:
+          forecast_period (on time coord): 4.0 hours
+     Scalar coordinates:
+          forecast_reference_time: 2015-11-23 03:00:00
+          threshold: 0.5 mm hr-1
+     Data:
+          All points contain float(1.) except the
+          zero point [0, 0, 7, 7] which is float(0.)
+"""
```

Comment:
Are all the docstring indentations something to do with sphinx rendering?  If not could you line these up please.  (Applies several times in this file.)